sdk/attestation/azure-security-attestation/src/attestation_common_request.cpp

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "private/attestation_common_request.hpp"
+
+#include "azure/core/exception.hpp"
+#include "azure/core/http/http.hpp"
+#include <memory>
+
+using namespace Azure::Security::Attestation;
+using namespace Azure::Security::Attestation::_detail;
+using namespace Azure::Core::Http::_internal;
+
+std::unique_ptr<Azure::Core::Http::RawResponse> AttestationCommonRequest::SendRequest(
+    Azure::Core::Http::_internal::HttpPipeline const& pipeline,
+    Azure::Core::Http::Request& request,
+    Azure::Core::Context const& context)
+{
+  auto response = pipeline.Send(request, context);
+  auto responseCode = response->GetStatusCode();
+
+  switch (responseCode)
+  {
+
+    // 200, 201, 202, 204 are accepted responses
+    case Azure::Core::Http::HttpStatusCode::Ok:
+    case Azure::Core::Http::HttpStatusCode::Created:
+    case Azure::Core::Http::HttpStatusCode::Accepted:
+    case Azure::Core::Http::HttpStatusCode::NoContent:
+      break;
+    default:
+      throw Azure::Core::RequestFailedException(response);
+  }
+  return response;
+}
+
+Azure::Core::Http::Request AttestationCommonRequest::CreateRequest(
+    Azure::Core::Url url,
+    Azure::Core::Http::HttpMethod method,
+    std::vector<std::string> const& path,
+    Azure::Core::IO::BodyStream* content)
+{
+  using namespace Azure::Core::Http;
+  return CreateRequest(url, "", method, path, content);
```

Comment:
Some APIs we send require an API version, others do not (because the standard for that API does not include an API version). Rather than have a "CreateRequest" and "CreateUnversionedRequest", I put the API version in each API when it's needed.